### ISSUE TITLE
no icon task bar no icon appmenu ksnip with appimage launcher

### ISSUE THREAD

**Issue by Lvaskz** [TheAssassin/AppImageLauncher#105]
Hi there, I have installed ksnip continuos appimage from here https://github.com/DamirPorobic/ksnip/releases/download/continuous/ksnip-1.5.0-continuous-x86_64.AppImage
Appimagelauncher install it but icon from appmenu and task bar are missing
I have linuxmit 19
cinnamon 
64 bit
with ubuntu 18.04 works well 
![ksnip icon missing task bar menu app](https://user-images.githubusercontent.com/29207169/49554097-59467880-f8c0-11e8-9e7b-246e93dfa05c.png)
![task bar](https://user-images.githubusercontent.com/29207169/49554135-7f6c1880-f8c0-11e8-9c49-74f28ec58dff.png)
If I run directly appimage in the application folder the icon on taskbar works well
![taskbar well](https://user-images.githubusercontent.com/29207169/49554214-c8bc6800-f8c0-11e8-98bb-be4399f9f15a.png)

**Comment by Lvaskz** [comment]
linuxmint info
![info linuxmint](https://user-images.githubusercontent.com/29207169/49554413-babb1700-f8c1-11e8-9536-bf2a6ed1f426.png)

**Comment by Lvaskz** [comment]
I installed appimagelauncher Linux mint 19.1 and same issue.

**Comment by Lvaskz** [comment]
hi, any progress?

**Comment by TheAssassin** [comment]
I wish I could dedicate time to this (IMO minor) issue right now. Please be a bit more patient. Your issue is not forgotten.

**Comment by Lvaskz** [comment]
ok, no problem. thanks

**Comment by TheAssassin** [comment]
This must be a problem in libappimage, something broke in there. The icon gets extracted as `.local/share/icons/hicolor/32x32/apps/appimagekit_500da126e82ba771ea90e1309c01dc34_ksnip`. This file lacks an extension, and it's the wrong resolution.

@azubieta can you please have a look? This must affect all AppImage products.

**Comment by TheAssassin** [comment]
@probonopd claims it works fine in appimaged, which uses a [4-months old version](https://github.com/AppImage/appimaged/tree/master/lib) of the library. So it must've broke in that time.

The issue seems not to be limited to `png` icons, like the one ksnip uses. I also build some custom AppImages for IDEs, etc., which don't contain additional icons in `usr/share/icons`. There must be some fallback code that extracts the AppDir root icon by default into `hicolor/32x32/apps`. I suppose that used to work fine... so I ask myself whether adding an extension to these files might fix the issue...

**Comment by TheAssassin** [comment]
It seems that the cleanup in the `unregister` method also failed to remove old files...

**Comment by TheAssassin** [comment]
Adding the extension fixes the issue.

**Comment by TheAssassin** [comment]
Issue identified, see link above.

**Comment by TheAssassin** [comment]
Please reopen if the issue persists.

**Comment by Lvaskz** [comment]
Thanks, It works now.

**Comment by Lvaskz** [comment]
Hi, It is me again. For reasons of vision I returned to ubuntu 18.04.02. I installed appimagelauncher_1.2.2-travis556.git20190414.ba13bec.bionic_amd64.deb and again integrated ksnip here https://github.com/DamirPorobic/ksnip/releases/download/continuous/ksnip-1.6.0-continuous-x86_64.AppImage, but the image in the  ubuntu dock is missing. The odd thing is that in linuxmint 19.1 is well displayed, the only detail in linuxmint is that the icon in the taskbar when launch is is missing initially, after closing the application and run it again that is fixed. I thought it was the same in ubuntu, but it's not working. I'm not sure why in some distributions it shows up well and in others it does not, I do not know if all distributions change the handling of these things in some way.
Might check?

**Comment by TheAssassin** [comment]
I guess you're using KDE again? Mind to log out and log in again? That usually clears the KDE icon caches.

**Comment by Lvaskz** [comment]
No kde, gnome desktop instead.Ubuntu 18.04.02 comes with gnome 3.30 by default

**Comment by TheAssassin** [comment]
Will have to try this myself. The ksnip AppImage seems okay, according to [appimagelint](https://github.com/TheAssassin/appimagelint).

**Comment by Lvaskz** [comment]
ok, thanks

**Comment by Lvaskz** [comment]
To continue with the issue, I made a new installation of kde neon 5.16.3 on my machine and installed appimagelauncher_1.3.1-travis702.git20190716.6b834d4.bionic_amd64.deb and installed ksnip-1.6.0-continuous-x86_64.AppImage and the icon image does not appear, despite restarting my laptop several times.
This problem appears in ubuntu 18.04, kde neon 5.16.3, kubuntu 18.04, but does not appear in linuxmint 19.1. I keep wondering if distributions handle this issue of icon storage differently in the system.

**Comment by Lvaskz** [comment]
I tried to add the icon manually from .local/share/icons.. etc etc , but by some reason that I unknown, the image(icon).png is not recognized. It appears , I can watch it , but when I add it is not recognized.
I did the same procedure with other appimages and it worked; as synfig, opentoonz morevna edition, parrot ng, but I did not achieve it with openshot or ksnip, for the same reason stated above.
There must be something strange in the folders where the icon images are stored. I don't know if the resolutions are the right ones, but I'm just speculating.
